### PR TITLE
store: fix waitgroup bug.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

**Comment by siddontang** [comment]
no test?

**Comment by qiuyesuifeng** [comment]
Seems it is hard to build a test.
I got this panic error 
```
PANIC: infoschema_test.go:37: testSuite.TestT

... Panic: sync: WaitGroup is reused before previous Wait has returned (PC=0x5A4CE)

/usr/local/go/src/runtime/panic.go:423
 in gopanic
/usr/local/go/src/sync/waitgroup.go:128
 in WaitGroup.Wait
/opt/cuiqiu/go/src/github.com/pingcap/tidb/store/localstore/compactor.go:187
```

I think this method is much safer.

**Comment by siddontang** [comment]
Got it.
LGTM

File: store/localstore/compactor.go [hunk 1]
```diff
@@ -178,6 +176,7 @@ func (gc *localstoreCompactor) Compact(ctx interface{}, k kv.Key) error {
 
 func (gc *localstoreCompactor) Start() {
 	// Start workers.
+	gc.workerWaitGroup.Add(deleteWorkerCnt + 1)
```
Comment (ngaut) [inline store/localstore/compactor.go]: Looks wired. Please use two wait group instead.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**